PalasoUIWindowsForms/ImageToolbox/ArtOfReadingChooser.cs

```diff
@@ -77,33 +86,32 @@ private void _searchButton_Click(object sender, EventArgs e)
 			try
 			{
 				_thumbnailViewer.Clear();
-				if (!string.IsNullOrEmpty(_searchTermsBox.Text))
+				if (!string.IsNullOrWhiteSpace(_searchTermsBox.Text))
 				{
 					bool foundExactMatches;
-					IEnumerable<object> results = _imageCollection.GetMatchingPictures(_searchTermsBox.Text, out foundExactMatches);
-					if (results.Count() == 0)
-					{
-						_messageLabel.Visible = true;
-						_searchResultStats.Text = "Found no matching images".Localize("ImageToolbox.NoMatchingImages");
-					}
-					else
+					// (avoid enumerating the returned IEnumerable<object> more than once by copying to a List.)
+					var results = new List<object>(_imageCollection.GetMatchingPictures(_searchTermsBox.Text, out foundExactMatches));
+					if (results.Any())
 					{
 						_messageLabel.Visible = false;
+						betterLinkLabel1.Visible = false;
```

Comment:
This variable needs a meaningful name. There is no clue here why we would want to hide it.